### PR TITLE
Task related to threading

### PR DESCRIPTION
**Pull request by vvaryshchuk**

### CODE REVIEW

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    names = ['na', 'ko', 'mi', 'fo', 'ad', 're', 'du', 'ly', 'ke', 'su', 'bo', 'na']
+    extensions = ['.txt', '.zip', '.ini', '.jpg', '.ttf']
+    file_names = []
+
+    for i in range(len_of_list):
+        file_name = ''
+        for j in range(3):
+            index = names.index(random.choice(names))
+            file_name += names[index]
+        ext = extensions.index(random.choice(extensions))
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: It's super complicated and useless block of code.

random.choice(extensions) returns you exactly one of the extensions. You don't need to get an index of this extension and extract it again using this index.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    """This function returns extension of file passing file name as parameter"""
+    _, extension = file_name.split('.')
+    return extension
+
+
+def generate_file_names_list():
+    """This function generate list of files with random names. Length of list is specified by global variable
+    len_of_list which value depend on number on thread."""
+    names = ['na', 'ko', 'mi', 'fo', 'ad', 're', 'du', 'ly', 'ke', 'su', 'bo', 'na']
+    extensions = ['.txt', '.zip', '.ini', '.jpg', '.ttf']
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: It's better to have a list of extensions without an extra dot.
You can join file name and extension using string's operation.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+def run_executors(list_of_file_names):
+    """Function which returns random size for file names from a list."""
+    for item in list_of_file_names:
+        threads = []
+        extension = recognize_file_type(item)
+
+        my_queue = Queue.Queue()
+
+        for i in range(0, number_of_threads):
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: You don't need this extra zero. Range works in this way by default.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def recognize_file_type(file_name):
+    """This function returns extension of file passing file name as parameter"""
+    _, extension = file_name.split('.')
+    return extension
+
+
+def generate_file_names_list():
+    """This function generate list of files with random names. Length of list is specified by global variable
```
[the remark this answers is not in the document]
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Ok, I modified this description.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Thanks, now I see my mistake. According to your remarks, I changed this part of the code.

Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: According to your remarks, I changed this part of the code.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 2 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Thanks for your advice. I removed dots from this list.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 3 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Sure, I understand it. Using zero in range() is my bad habit. I'll try to omit it next time.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+"""
+
+import random
+import string
+import Queue
+from threading import Thread
+
+
+number_of_threads = 5
+len_of_list = number_of_threads * 100
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Bad name of the variable.
It tells about nothing.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+def run_executors(list_of_file_names):
+    """Function which returns random size for file names from a list."""
+    for item in list_of_file_names:
+        threads = []
+        extension = recognize_file_type(item)
+
+        my_queue = Queue.Queue()
+
+        for i in range(number_of_threads):
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: It looks like for each file type you'll start size_of_files * 100 threads what is not expected.

Example:

data = ['test.txt', 'image.jpg']

Your code will start 200 threads for processing `test.txt` and 200 more threads for `image.jpg`.

**Comment by pivancy** [comment]
Why are you trying to remove your previous tasks from the repository?

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 5 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Ok. I renamed this variable.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 6 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: You are right. I fixed it by moving some functionality in other function and then run this function in separate threads. These changes are included in the last commit. Is it good or I should look for another solution?

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    """
+    This function generate list of files with random names.
+    Length of list is specified by global variable
+    len_of_list which value depend on number of thread.
+    """
+    extensions = ['txt', 'zip', 'ini', 'jpg', 'ttf']
+    file_names = []
+    for i in range(length_of_list):
+        file_name = ''.join([random.choice(string.ascii_lowercase) for j in range(5)])
+        full_file_name = '.'.join([file_name, random.choice(extensions)])
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: It's not clear from the name what is `full_file_name`.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+This module returns random size of file name from list by processing elements in separate threads
+"""
+
+import random
+import string
+from threading import Thread
+
+
+number_of_threads = 3
+length_of_list = number_of_threads * 100
```
Comment (pivancy) [inline vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: `length_of_list` is a bad name of variable.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 7 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: Ok, I changed this name in last commit. I think now it looks more clearly.

File: vasyl_varyshchuk/Week_2/task_1/func_in_threads.py (hunk 8 above)
Comment (vvaryshchuk) [reply vasyl_varyshchuk/Week_2/task_1/func_in_threads.py]: As in previous case, I modified name of variable and added this change to my last commit.

**Comment by pivancy** [comment]
@vvaryshchuk , why did you merge this pull request?

Did you see 👍 symbol here?

Not all of my questions has your answers here

**Comment by vvaryshchuk** [comment]
Sorry for this mistake. I guessed that I can merge after your approove. Currently I reverted master branch to state before merge. Is it possibly to reopen this pull request or I should create new one?

**Pull request merged**